rootclock.c

```diff
@@ -16,17 +17,37 @@
 #include "drw.h"
 #include "util.h"
 
+static volatile sig_atomic_t running = 1;
+
+static void
+signal_handler(int sig)
+{
+  (void)sig; /* unused parameter */
+  running = 0;
+}
+
 static void draw_block_for_region(Drw *drw, int rx, int ry, int rw, int rh,
                                   Fnt *tf, Fnt *df, int show_date_flag,
                                   Clr *bg_scm, Clr *time_scm, Clr *date_scm,
                                   const char *tstr, const char *dstr,
                                   int block_yoff, int spacing) {
+  /* Basic parameter validation */
+  if (!drw || !tf || !bg_scm || !time_scm || !tstr) {
+    return;
+  }
+  if (show_date_flag && (!df || !date_scm)) {
+    return;
+  }
+  
```

Comment:
Combine these checks into a single condition.
Print an error to stderr to make it easier for users to understand what went wrong.